src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,116 @@
+
+    @Override
+    public Optional<Product> get(Long id) {
+        String query = "SELECT * FROM PRODUCTS WHERE ID = ? AND DELETED = false;";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                String name = resultSet.getString("NAME");
+                double price = resultSet.getDouble("PRICE");
+                Product product = new Product(name, price);
+                product.setId(id);
+                return Optional.of(product);
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting product by id="
+                    + id + " is failed", e);
+        }
+    }
+
+    @Override
+    public List<Product> getAll() {
+        String query = "SELECT * FROM PRODUCTS WHERE DELETED = false;";
+        List<Product> products = new ArrayList<>();
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                long id = resultSet.getLong("ID");
+                String name = resultSet.getString("NAME");
+                double price = resultSet.getDouble("PRICE");
+                Product product = new Product(name, price);
+                product.setId(id);
+                products.add(product);
+            }
+            return products;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get product from DB", e);
+        }
+    }
+
+    @Override
+    public Product update(Product product) {
+        String query = "UPDATE PRODUCTS SET NAME = ?, PRICE = ? WHERE ID = ? AND DELETED = false;";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, product.getName());
+            statement.setDouble(2, product.getPrice().doubleValue());
+            statement.setLong(3, product.getId());
+            statement.executeUpdate();
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Updating product with id="
+                    + product.getId() + " is failed", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE PRODUCTS SET DELETED = true WHERE ID = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            statement.executeUpdate();
+            return true;
```

Comment:
should we always return true? let's check the result of statement.executeUpdate() and return the result of checking